packages/rooks/src/hooks/useDocumentEventListener.ts

```diff
@@ -18,14 +18,20 @@ function useDocumentEventListener(
   listenerOptions: ListenerOptions = {},
   isLayoutEffect = false
 ): void {
-  useGlobalObjectEventListener(
-    document,
-    eventName,
-    callback,
-    listenerOptions,
-    true,
-    isLayoutEffect
-  );
+  if (typeof document !== "undefined") {
+    /*
+    Since the above condition changes values only across different environments, it is fine to call the hook conditionally
+    */
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useGlobalObjectEventListener(
+      document,
```

Comment:
@gmahima  I just got this idea, instead of using a conditional, can you try just using `global.document` here?

Reply:
> @gmahima I just got this idea, instead of using a conditional, can you try just using `global.document` here?

It's working! It's working for the other hooks which use `window` too.

Reply:
But how is it working? Why is it not throwing an error? Isn't `global.window` and `window` the same object?

Reply:
It works because the reference it looks for is now `global` and if `global.window` doesn't exist, it will return undefined, instead of throwing an error.

Reply:
@imbhargav5 oh! ok